newIDE/app/src/MainFrame/index.js

```diff
@@ -1188,6 +1189,32 @@ class MainFrame extends React.Component<Props, State> {
     }
   };
 
+  saveAs = () => {
+    saveUiSettings(this.state.editorTabs);
+
+    const { currentProject } = this.state;
+    if (!currentProject) return;
+    const { i18n } = this.props;
+
+    if (this.props.saveDialog) {
+      this._openSaveDialog();
+    } else if (this.props.onSaveProjectAs) {
+      this.props.onSaveProjectAs(currentProject).then(
+        () => {
```

Comment:
We'll add a boolean to show or not the snack. We don't want to show it if the user cancelled (see also my other comments about handling this boolean):
```
// ...
(saveDone) => {
  if (saveDone) this._showSnackMessage(i18n._(t`Project properly saved`));
}, //...
```